File: src/1.Arrays_and_hashing/4.Group_anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, List<string>> dict = new Dictionary<string, List<string>>();
        foreach(string str in strs)
        {
            char[] charArray = str.ToCharArray();
            Array.Sort(charArray);
            string key = new string(charArray);
            if(dict.ContainsKey(key))
            {
                dict[key].Add(str);
            }
            else
            {
                dict.Add(key, new List<string>{str});
            }
        }
        IList<IList<string>> result = new List<IList<string>>();
        foreach(KeyValuePair<string, List<string>> entry in dict)
        {
            result.Add(entry.Value);
        }
        return result;
    }
}

File: src/1.Arrays_and_hashing/4.Group_anagrams.py
# https://leetcode.com/problems/group-anagrams/

def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
    dict = {}
    for str in strs:
        charArray = list(str)
        charArray.sort()
        key = "".join(charArray)
        if key in dict:
            dict[key].append(str)
        else:
            dict[key] = [str]
    result = []
    for key in dict:
        result.append(dict[key])
    return result